models/coded_test.go

```diff
@@ -0,0 +1,27 @@
+package models
+
+import "testing"
+
+func TestIntersection(t *testing.T) {
+	if len(computeIntersection([]string{"a", "b"}, []string{"a"})) != 1 {
+		t.Error("Incorrect number of intersecting elements")
+	}
+	if len(computeIntersection([]string{"a", "b"}, []string{"a", "b"})) != 2 {
+		t.Error("Incorrect number of intersecting elements")
+	}
+	if len(computeIntersection([]string{"a", "b"}, []string{"c", "d"})) != 0 {
+		t.Error("Incorrect number of intersecting elements")
+	}
+}
+
+func TestPreferredCode(t *testing.T) {
```

Comment:
I'd put this in a file called `entry_test.go`, just because the `PreferredCode` method is defined in `entry.go`